ASM/c/item_table.c

```diff
@@ -309,22 +309,67 @@ item_row_t item_table[] = {
     0x16 messages after the base message.
 
     Returns: text_id to use based on the logic above.
-*/ 
+*/
+
+uint16_t resolve_text_silver_rupees(item_row_t *item_row, bool is_outgoing) {
+    // Get the arguments from the item_row struct.
+    int16_t dungeon_id = item_row->effect_arg1;
+    int16_t silver_rupee_id = item_row->effect_arg2;
+
+    // Get the data for this silver rupee puzzle
+    silver_rupee_data_t var = silver_rupee_vars[silver_rupee_id][CFG_DUNGEON_IS_MQ[dungeon_id]];
+
+    if (!is_outgoing && extended_savectx.silver_rupee_counts[silver_rupee_id] >= var.needed_count) {
+        // The player already has enough silver rupees.
+        return item_row->text_id + 0x16;
+    } else if (!is_outgoing && extended_savectx.silver_rupee_counts[silver_rupee_id] + 1 == var.needed_count) {
+        // This silver rupee unlocks the puzzle.
+        return item_row->text_id + (CFG_DUNGEON_IS_MQ[dungeon_id] ? 0x2E : 0x44);
+    } else if (!is_outgoing && extended_savectx.silver_rupee_counts[silver_rupee_id] > 0) {
+        // Show number collected.
+        return item_row->text_id;
+    } else {
+        // This silver rupee is for another player or the first of this puzzle, don't show any extra info.
```

Comment:
Don't think I agree with this change. I think it's more consistent to just always show the count if it's for this player.

Reply:
This is how it works for small keys as well.

Reply:
Yeah I noticed. Thought it was broken until I collected a second one...